lib/client/mediator-subscribers/list.js

```diff
@@ -21,18 +21,24 @@ module.exports = function subscribeToListTopic(syncDatasetTopics, datasetManager
     var self = this;
     parameters = parameters || {};
 
-    //Creating the item in the sync store
-    datasetManager.list().then(function(arrayOfDatasetItems) {
-      var listDoneTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.LIST, CONSTANTS.DONE_PREFIX, parameters.topicUid);
 
-      self.mediator.publish(listDoneTopic, arrayOfDatasetItems);
-
-    }).catch(function(error) {
-
-      var listErrorTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.LIST, CONSTANTS.ERROR_PREFIX, parameters.topicUid);
-
-      self.mediator.publish(listErrorTopic, error);
+    datasetManager.getMetaData(function(metadata) {
+      datasetManager.list().then(function(arrayOfDatasetItems) {
+        var listDoneTopic = syncDatasetTopics.getTopic(CONSTANTS.TOPICS.LIST, CONSTANTS.DONE_PREFIX, parameters.topicUid);
+
+        if (metadata.syncEvents) {
+          arrayOfDatasetItems.forEach(function(item) {
```

Comment:
Probably use lodash here as arrayOfDatasetItems may be null and calling `.forEach` will cause an uncaught exception